ui/src/api-logic/mappers/users.ts

```diff
@@ -0,0 +1,38 @@
+import { UserData } from 'src/@types'
+import { USER_TYPE } from 'src/utils/constants'
+import { UsersQuery } from '../graphql/generated/schema'
+
+/**
+ * Method which iterate each user and map it to UserData to be shown in table
+ * @param data User list that comes from data base in UserData format
+ * @returns User list from data base mapped to RepoImportData list
+ */
+const mapToUserData = (data: UsersQuery | undefined): Array<UserData> => {
+  const mappedData: Array<UserData> = []
+
+  data?.userMgmtPgUsers?.nodes.forEach((usr) => {
+    if (usr.memberof && usr.memberof.length > 0) {
+      // Consolidated User info
+      const userInfo: UserData = {
+        name: usr.rolname || '',
+        roles: usr.memberof.map(getRole)
+      }
+      mappedData.push(userInfo)
+    }
+  })
+
+  return mappedData
+}
+
+const getRole = (role: string | null) => {
+  switch (role) {
+    case 'mergestat_role_admin':
+      return USER_TYPE.ADMIN
+    case 'mergestat_role_user':
+      return USER_TYPE.USER
+    default:
+      return USER_TYPE.READ_ONLY
+  }
```

Comment:
I think we should be careful about defaulting to the `READ_ONLY` role here - since it means something specific. If a role cannot be determined, I think we should display `Unknown` in the UI (and maybe have that as a type here).

Reply:
Done [5c1d507](https://github.com/mergestat/mergestat/pull/556/commits/5c1d507283340088fdb99a37e8fdbab9a7320ade)